datastore/providers/mongodb_atlas_datastore.py

```diff
@@ -0,0 +1,265 @@
+import arrow
+import os
+from typing import Dict, List, Any, Optional
+from loguru import logger
+from bson.objectid import ObjectId
+from importlib.metadata import version
+from motor.motor_asyncio import AsyncIOMotorClient
+from pymongo.driver_info import DriverInfo
+
+from datastore.datastore import DataStore
+from functools import cached_property
+from models.models import (
+    DocumentChunk,
+    DocumentChunkWithScore,
+    DocumentMetadataFilter,
+    QueryResult,
+    QueryWithEmbedding,
+)
+from services.date import to_unix_timestamp
+
+
+MONGODB_CONNECTION_URI = os.environ.get("MONGODB_URI")
+MONGODB_DATABASE = os.environ.get("MONGODB_DATABASE", "default")
+MONGODB_COLLECTION = os.environ.get("MONGODB_COLLECTION", "default")
+MONGODB_INDEX = os.environ.get("MONGODB_INDEX", "default")
+OVERSAMPLING_FACTOR = 10
+MAX_CANDIDATES = 10_000
+UPSERT_BATCH_SIZE = 100
```

Comment:
UPSERT_BATCH_SIZE is never used. Does it even need to be used since insert_many already batches optimally?

Reply:
Good catch. I'll remove this. I only just saw that you reviewed this.